### PR TITLE
Updated the course & book resources w AI additions

### PR DESCRIPTION
**Pull request by WaunBroderick**
check_urls=free-courses-en.md free-programming-books-en.md

## What does this PR do?
Add resource(s) | Remove resource(s) | Add info | Improve repo

## For resources
Free courses and free programming books in the emergent and very popular field of artificial intelligence and graph neural networks.

### Why is this valuable (or not)?
I believe this is one of the new blossoming topics in AI and finding resources for this is still difficult though becoming easier as the field exapnds. A lot of these books are still currently pre-published due to how new the knowledge is.

### How do we know it's really free?
It is shared in the open commons of McGill and Standford univeristy

### For book lists, is it a book? For course lists, is it a course? etc.
It is a book and a course the book is a series of PDFs, and for the course all of the pdf slides and collab docs are open to all.

## Checklist:
- [X] Read our [contributing guidelines](https://github.com/EbookFoundation/free-programming-books/blob/main/docs/CONTRIBUTING.md).
- [X] [Search](https://ebookfoundation.github.io/free-programming-books-search/) for duplicates.
- [X] Include author(s) and platform where appropriate.
- [X] Put lists in alphabetical order, correct spacing.
- [X] Add needed indications (PDF, access notes, under construction).
- [X] Used an informative name for this pull request.

## Follow-up

- Check the status of GitHub Actions and resolve any reported warnings!

### CODE REVIEW

File: books/free-programming-books-subjects.md [hunk 1]
```diff
@@ -134,12 +134,14 @@ Books that cover a specific programming language can be found in the  [BY PROGRA
...
 * [Artificial Intelligence: Foundations of Computational Agents (2017), 2nd Edition](https://artint.info) - David L. Poole, Alan K. Mackworth @ Cambridge University Press (HTML, Slides)
+* [Graph Representational Learning](https://www.cs.mcgill.ca/~wlh/grl_book/) - William L. Hamilton, McGill University (HTML, PDF)
 * [Introduction to Autonomous Robots](https://github.com/correll/Introduction-to-Autonomous-Robots/releases) - Nikolaus Correll (PDF)
 * [On the Path to AI: Law’s prophecies and the conceptual foundations of the machine learning age](https://link.springer.com/book/10.1007/978-3-030-43582-0) - Thomas D. Grant, Damon J. Wischik (PDF)
 * [Probabilistic Programming & Bayesian Methods for Hackers](http://camdavidsonpilon.github.io/Probabilistic-Programming-and-Bayesian-Methods-for-Hackers/) - Cam Davidson-Pilon (HTML, Jupyter Notebook)
 * [The Quest for Artificial Intelligence: A History of Ideas and Achievements](http://ai.stanford.edu/~nilsson/QAI/qai.pdf) - Nils J. Nilsson (PDF)
 
 
+
```
Comment (metabiswadeep) [inline books/free-programming-books-subjects.md:144]:
```suggestion
```

**Review by metabiswadeep — CHANGES_REQUESTED** [review]

File: books/free-programming-books-subjects.md (hunk 1 above)
Comment (metabiswadeep) [reply books/free-programming-books-subjects.md:144]: Remove useless space

File: courses/free-courses-en.md [hunk 2]
```diff
@@ -198,6 +198,7 @@
 * [IBM AI Engineering Professional Certificate](https://www.coursera.org/professional-certificates/ai-engineer) - Romeo Kienzler, Saeed Aghabozorgi, Joseph Santarcangelo, Alex Aklson et al. (Coursera)
 * [MIT Deep Learning and Artificial Intelligence Lectures](https://deeplearning.mit.edu) - Lex Fridman, et al.
 * [MIT's Artificial Intelligence](http://ocw.mit.edu/courses/electrical-engineering-and-computer-science/6-034-artificial-intelligence-fall-2010/) - Prof. Patrick Henry Winston (MIT OpenCourseWare)
+* [Stanford's Machine Learning with Graphs](https://web.stanford.edu/class/cs224w/) - Prof. Jurij Leskovac (Stanford University)
```
Comment (eshellman) [inline courses/free-courses-en.md:201]: we don't list homorifics, such as "Prof."
we don't list author affiliations either.

File: books/free-programming-books-subjects.md [hunk 3]
```diff
@@ -134,6 +134,7 @@ Books that cover a specific programming language can be found in the  [BY PROGRA
 * [Artificial Intelligence for a Better Future: An Ecosystem Perspective on the Ethics of AI and Emerging Digital Technologies](https://link.springer.com/book/10.1007/978-3-030-69978-9) - Bernd Carsten Stahl (PDF)
 * [Artificial Intelligence: Foundations of Computational Agents (2010), 1st Edition](https://artint.info/aifca1e.html) - David L. Poole, Alan K. Mackworth @ Cambridge University Press (HTML)
 * [Artificial Intelligence: Foundations of Computational Agents (2017), 2nd Edition](https://artint.info) - David L. Poole, Alan K. Mackworth @ Cambridge University Press (HTML, Slides)
+* [Graph Representational Learning](https://www.cs.mcgill.ca/~wlh/grl_book/) - William L. Hamilton, McGill University (HTML, PDF)
```
Comment (eshellman) [inline books/free-programming-books-subjects.md:137]: remove affiliation

**Review by eshellman — CHANGES_REQUESTED** [review]

File: courses/free-courses-en.md (hunk 2 above)
Comment (davorpa) [reply courses/free-courses-en.md:201]:
```suggestion
* [Stanford's Machine Learning with Graphs](https://web.stanford.edu/class/cs224w/) - Jurij Leskovac (Stanford University)
```

File: books/free-programming-books-subjects.md (hunk 3 above)
Comment (davorpa) [reply books/free-programming-books-subjects.md:137]:
```suggestion
* [Graph Representational Learning](https://www.cs.mcgill.ca/~wlh/grl_book/) - William L. Hamilton (HTML, PDF)
```

**Comment by WaunBroderick** [comment]
@eshellman Thanks Eric, I believe I made all the necessary changes as requested, though seems GitHub thinks one is left? IF you could correct me if i'm missing something I'd appreciate it, and thanks in advance!

**Review by WaunBroderick — COMMENTED** [review]
should be good to go

**Review by WaunBroderick — COMMENTED** [review]
changed

**Pull request closed without merging**